src/heroic.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::HashMap;
+
+use crate::{
+    config::RootsConfig,
+    manifest::{Manifest, Store},
+    prelude::{normalize_title, StrictPath},
+};
+
+// Deserialization of Heroic gog_store/installed.json
+#[derive(serde::Deserialize)]
+struct HeroicInstalledGame {
+    #[serde(rename = "appName")]
+    game_id: String,
+    platform: String,
+}
+#[derive(serde::Deserialize)]
+struct HeroicInstalled {
+    installed: Vec<HeroicInstalledGame>,
+}
+
+// Deserialization of Heroic gog_store/library.json
+#[derive(serde::Deserialize)]
+struct GogLibraryGame {
+    app_name: String,
+    title: String,
+}
+#[derive(serde::Deserialize)]
+struct GogLibrary {
+    games: Vec<GogLibraryGame>,
+}
+
+// Deserialization of Legendary legendary/installed.json
+#[derive(serde::Deserialize)]
+struct LegendaryInstalledGame {
+    #[serde(rename = "app_name")]
+    game_id: String,
+    title: String,
+    platform: String,
+}
+#[derive(serde::Deserialize)]
+struct LegendaryInstalled(HashMap<String, LegendaryInstalledGame>);
+
+// TODO.2022-10-10 heroic: windows location for legendary
+// TODO.2022-10-15 heroic: make relative to heroic root in question!
```

Comment:
Does this comment just apply to the Flatpak path?

Reply:
The relative path? On my systems it applies to both kinds of installations.  Normal is `~/.config/heroic` + `~/.config/legendary` and flatpak is `~/.var/app/com.heroicgameslauncher.hgl/config/heroic` + `~/.var/app/com.heroicgameslauncher.hgl/config/legendary`

Since heroic refers to it's sibbling legendary installation a relative path seems to be a good idea to make sure ludusavi finds the correct one.

Reply:
Gotcha. If we do a relative/sibling check, we would still need the absolute path as a fallback. At least on Windows, they're not sibling folders. We can handle this later in #142 and #144.